### PR TITLE
Headers cleanup

### PR DESCRIPTION
**Pull request by benmccann**
This isn't working yet, but I thought I'd share for feedback on the direction

I introduced a `Headers` class as it was too annoying to deal with values that could be `string|string[]` everywhere and converting between all the different headers types

### CODE REVIEW

**Comment by Rich-Harris** [comment]
I wonder if we're overcomplicating this? According to the 'message.headers' section of https://nodejs.org/api/http.html#http_class_http_incomingmessage, the only time a header could be an array of strings is `set-cookie`. And the only time `set-cookie` could be involved is when `prepare` or an endpoint is populating response headers (it doesn't make sense in the context of request headers, even if the type definitions don't distinguish between request and response headers).

In other words I think we might be able to add 

```ts
type RequestHeaders = {
  [key: string]: string
};

type ResponseHeaders = {
  'set-cookie': string | string[];
  [key: string]: string;
};
```

and call it a day. We'd need some logic for merging headers (using the logic articulated in the link above) but as far as the types go I think we can probably make this quite simple.

Looking into this now

**Comment by benmccann** [comment]
Closing in favor of https://github.com/sveltejs/kit/pull/445
